web/extensions/core/maskeditor.js

```diff
@@ -0,0 +1,542 @@
+			imgCanvas.height = window.innerHeight - 200;
+
+			// redraw image
+			let drawWidth = orig_image.width;
+			let drawHeight = orig_image.height;
+			if (orig_image.width > imgCanvas.width) {
+				drawWidth = imgCanvas.width;
+				drawHeight = (drawWidth / orig_image.width) * orig_image.height;
+			}
+
+			if (drawHeight > imgCanvas.height) {
+				drawHeight = imgCanvas.height;
+				drawWidth = (drawHeight / orig_image.height) * orig_image.width;
+			}
+
+			imgCtx.drawImage(orig_image, 0, 0, drawWidth, drawHeight);
+
+			// update mask
+			backupCtx.drawImage(maskCanvas, 0, 0, maskCanvas.width, maskCanvas.height, 0, 0, backupCanvas.width, backupCanvas.height);
+			maskCanvas.width = drawWidth;
+			maskCanvas.height = drawHeight;
+			maskCanvas.style.top = imgCanvas.offsetTop + "px";
+			maskCanvas.style.left = imgCanvas.offsetLeft + "px";
+			maskCtx.drawImage(backupCanvas, 0, 0, backupCanvas.width, backupCanvas.height, 0, 0, maskCanvas.width, maskCanvas.height);
+		});
+
+		const filepath = ComfyApp.clipspace.images;
+
+		const touched_image = new Image();
+
+		touched_image.onload = function() {
+			backupCanvas.width = touched_image.width;
+			backupCanvas.height = touched_image.height;
+
+			removeRGB(touched_image, backupCanvas, backupCtx, maskCtx);
+		};
+
+		touched_image.src = ComfyApp.clipspace.imgs[0].src;
+
+		// original image load
+		orig_image.onload = function() {
+			window.dispatchEvent(new Event('resize'));
+		};
+
+		orig_image.src = ComfyApp.clipspace.original_imgs[0].src;
+		this.image = orig_image;
+	}
+
+
+	setEventHandler(maskCanvas) {
+		maskCanvas.addEventListener("contextmenu", (event) => {
+			event.preventDefault();
+		});
+
+		const self = this;
+		maskCanvas.addEventListener('wheel', (event) => this.handleWheelEvent(self,event));
+		maskCanvas.addEventListener('mousedown', (event) => this.handleMouseDown(self,event));
+		document.addEventListener('mouseup', MaskEditorDialog.handleMouseUp);
+		maskCanvas.addEventListener('mousemove', (event) => this.draw_move(self,event));
+		maskCanvas.addEventListener('touchmove', (event) => this.draw_move(self,event));
+		document.addEventListener('keyup', MaskEditorDialog.handleKeyUp);
+	}
+
+	brush_size = 10;
+	drawing_mode = false;
+	lastx = -1;
+	lasty = -1;
+	lasttime = 0;
+
+	static handleKeyUp(event) {
+		const self = MaskEditorDialog.instance;
+		if (event.key === ']') {
+			self.brush_size = Math.min(self.brush_size+2, 100);
+		} else if (event.key === '[') {
+			self.brush_size = Math.max(self.brush_size-2, 1);
+		}
+	}
+
+	static handleMouseUp(event) {
+		event.preventDefault();
+		MaskEditorDialog.instance.drawing_mode = false;
+	}
+
+	handleWheelEvent(self, event) {
+		if(event.deltaY < 0)
+			self.brush_size = Math.min(self.brush_size+2, 100);
+		else
+			self.brush_size = Math.max(self.brush_size-2, 1);
+
+
+		self.brush_slider_input.value = self.brush_size;
+//		self.brush_slider_input.dispatchEvent(new Event('input'));
```

Comment:
```suggestion
//		self.brush_slider_input.dispatchEvent(new Event('input'));

        const brush = this.brush;
        brush.style.width = (this.brush_size - 1) * 2 + "px";
        brush.style.height = (this.brush_size - 1) * 2 + "px";
        brush.style.left = (event.pageX - this.brush_size) + "px";
        brush.style.top = (event.pageY - this.brush_size) + "px";
```
Change cursor with mousewheel.

Reply:
Oh.. I found that, we missed slider event and key event.